contracts/LifCrowdsale.sol

```diff
@@ -191,18 +210,41 @@ contract LifCrowdsale is Ownable, Pausable {
 
     uint256 tokens = weiSent.mul(privatePresaleRate);
 
-    require(totalPresaleWei.add(weiSent) <= maxPresaleWei);
-
     totalPresaleWei.add(weiSent);
 
     token.mint(beneficiary, tokens);
   }
 
   // send ether to the fund collection wallet
-  function forwardFunds() onlyOwner {
-    foundationWallet.transfer(this.balance);
-    // TODO
-    // marketMaker.transfer(marketMakerPercentage * this.balance);
+  function forwardFunds() internal {
+
+    // calculate the max amount of wei for the foundation
+    uint256 foundationBalanceCapWei= foundationBalanceCapUSD.mul(weiPerUSD);
+
+    // if the minimiun cap for the market maker is not reached transfer all funds to foundation
+    // else if the min cap for the market maker is reached, create it and send the remaining funds
+    if (this.balance < foundationBalanceCapWei) {
+
+      foundationWallet.transfer(this.balance);
+
+    } else {
+
+      uint256 mmFundBalance = this.balance.sub(foundationBalanceCapWei);
+
+      // check how much preiods we have to use on the market maker
+      uint8 marketMakerPeriods = 24;
+      if (mmFundBalance > marketMaker24PeriodsCapUSD.mul(weiPerUSD))
+        marketMakerPeriods = 48;
+
+      foundationWallet.transfer(foundationBalanceCapWei);
+      LifMarketMaker marketMaker = new LifMarketMaker(
+        address(token), block.number.add(10), 20, marketMakerPeriods, foundationWallet, 105000
+      );
+      marketMaker.fund.value(mmFundBalance)();
+
+      MMCreated(marketMaker);
```

Comment:
shouldn't we also have a `marketMaker` variable in the contract, that we set here, that gives an easy way to get to the marketMaker?

Reply:
We dont need to access the marketMaker from the crowdsale, just create it, thats why I only added an event. I think we can store the address if we need to access it, but since we dont an event in enough, once the crowdsale contract ends it wont be used any more.

Reply:
I mean users will know the address of the crowdsale stop they could go back to the crowdsale and take the market maker address from there. I'm not sure but it seems to me that extracting the value from the event is more complicated for users